cpp/src/lance/arrow/type.cc

```diff
@@ -161,16 +167,34 @@ ::arrow::Result<std::shared_ptr<::arrow::DataType>> FromLogicalType(
     return ::arrow::fixed_size_binary(size);
   }
 
-  if (logical_type.starts_with("dict")) {
+  if (logical_type.starts_with("extension")) {
+    // here logical_type is in the format extension:<extension_name>
+    // Note that an ExtensionType subclass instance with the extension_name
+    // must already be registered
     auto components = ::arrow::internal::SplitString(logical_type, ':');
-    if (components.size() != 4) {
+    if (components.size() != 2) {
+      return ::arrow::Status::Invalid(
+          fmt::format("Invalid extension type string: {}", logical_type.to_string()));
+    }
+    auto ext_type = ::arrow::GetExtensionType(components[1].to_string());
```

Comment:
not sure if this works for parametric types actually. I'll add a separate test for that.

Reply:
ok extension_name doesn’t work differently for parametric type. The only difference is that it’s possible for a parametric type to have different extension names based on the parameter, and it must be registered separately. But looking it up by the extension name is still the same.